tools/alara_output_processing/alara_output_processing.py

```diff
@@ -359,15 +369,85 @@ def _single_element_all_nuclides(self, element):
                 self.
 
         Returns:
-            nuclide_list (list): List of all nuclides of the selected element
+            nuclides (set): Set of all nuclides of the selected element
                 present in the ALARADFrame.
         '''
 
-        return self.loc[
+        return set(self.loc[
             self['nuclide'].str.startswith(f'{element.lower()}-', na=False),
             'nuclide'
-        ].unique().tolist()
+        ])
+    
+    @staticmethod
+    def _select_radionucs(filters):
+        '''
+        Boolean function to determine whether or not a list of filters in the
+            optional "half_life" key of filter_dicts is seeking unstable
+            nuclides.
+        
+        Arguments:
+            filters (list): List of filters provided for the "half_life" key.
+        
+        Returns:
+            select_radionucs (bool): True if the 0th element of 
+                filter_dict["half_life"] = "radioactive", "unstable", or any
+                of the Python mathematical operators (">", "<", ">=", "<=",
+                "==", "!="), otherwise False.
+        '''
+
+        return filters[0] in {'radioactive', 'unstable'} | set(OPS)
+    
+    @staticmethod
+    def _half_life_numeric_ops(expression, half_lives):
+        '''
+        Create a subset of a pre-existing set of half-lives contained in an
+            ALARADFrame that match the mathematical operation and threshold
+            provided in the partial expression provided (i.e. all half-lives
+            greater than 1e6 seconds)
+
+        Arguments:
+            expression (array-like): Array-like collection of a partial
+                mathematical expression of the form:
+                    expression[0] (str): String of a Python-formatted
+                    mathematical operator (">", "<", ">=", "<=", "==", "!=").
+                    expression[1] (int or float): threshold comparison value.
+                Together, the expression would be of the form ['>', 1e6].
+            half_lives (set): Set of all numeric half-lives in the ALARADFrame
+                (i.e. excluding "stable" for stable nuclides).
+
+        Returns:
+            half_lives (set): Subset of the input half_lives that match the
+                expression
+        '''
+
+        op, threshold = expression
+        return {hl for hl in half_lives if OPS[op](hl, threshold)}
     
+    def _get_half_lives(self, filters):
+        '''
+        Create a set of all numeric half-lives in an ALARADFrame.
+
+        Arguments:
+            self (alara_output_processing.ALARADFrame): Specialized ALARA
+                output DataFrame.
+        
+        Returns:
+            half_lives (set): Set of all numeric half-lives in the ALARADFrame
+                (i.e. excluding "stable" for stable nuclides).
+        '''
+
+        half_lives =  set(self.loc[
+            self['half_life'].apply(lambda x: isinstance(x, float)),
+            'half_life'
+        ])
```

Comment:
While this isn't too complicated, it makes we wonder if it isn't better to use some special float to indicate stable nuclides, e.g. -1?  Then this could just be  a simpler filter for `half_life >= 0`?